core/services/relay/evm/functions/config_poller.go

```diff
@@ -0,0 +1,192 @@
+package functions
+
+import (
+	"context"
+	"database/sql"
+	"encoding/binary"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/libocr/gethwrappers2/ocr2aggregator"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+var _ ConfigPoller = &configPoller{}
+
+type ConfigPoller interface {
+	ocrtypes.ContractConfigTracker
+
+	Start()
+	Close() error
+	Replay(ctx context.Context, fromBlock int64) error
+}
+
+type FunctionsPluginType int
+
+const (
+	FunctionsPlugin FunctionsPluginType = iota
+	ThresholdPlugin
+	S4Plugin
+)
+
+type configPoller struct {
+	lggr               logger.Logger
+	filterName         string
+	destChainLogPoller logpoller.LogPoller
+	addr               common.Address
```

Comment:
As a general note: let's avoid cryptic names. Like `addr` is unclear what is for, unless I know the domain.

Reply:
This code is copied from elsewhere, but that is a good note.